src/app/ui/editor/moving_slice_state.cpp

```diff
@@ -48,34 +56,271 @@ MovingSliceState::MovingSliceState(Editor* editor,
     }
   }
 
+  editor->getSite(&m_site);
```

Comment:
When the selected layers are a mix of tilemap and image layers + if the last layer chosen in the timeline is a Image Layer --> the ColorBar does not have the TilesetMode buttons and the code assumes 'Manual' at this point. This causes a loss of drawing if the slice is moved outside the original area where the tiles were drawn in the tilemap layer.
```suggestion
  editor->getSite(&m_site);
  if (!m_site.layer()->isTilemap()) {
    m_site.tilemapMode(TilemapMode::Pixels);
    m_site.tilesetMode(TilesetMode::Auto);
  }
```